pandas/tests/indexes/multi/test_compat.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+
+import numpy as np
+import pandas.util.testing as tm
+import pytest
+from pandas import MultiIndex
+from pandas.compat import PY3, long
+
+
+def test_numeric_compat(idx):
+    tm.assert_raises_regex(TypeError, "cannot perform __mul__",
+                           lambda: idx * 1)
+    tm.assert_raises_regex(TypeError, "cannot perform __rmul__",
+                           lambda: 1 * idx)
+
+    div_err = "cannot perform __truediv__" if PY3 \
+        else "cannot perform __div__"
+    tm.assert_raises_regex(TypeError, div_err, lambda: idx / 1)
+    div_err = div_err.replace(' __', ' __r')
+    tm.assert_raises_regex(TypeError, div_err, lambda: 1 / idx)
+    tm.assert_raises_regex(TypeError, "cannot perform __floordiv__",
+                           lambda: idx // 1)
+    tm.assert_raises_regex(TypeError, "cannot perform __rfloordiv__",
+                           lambda: 1 // idx)
+
+
+def test_logical_compat(idx):
+    tm.assert_raises_regex(TypeError, 'cannot perform all',
+                           lambda: idx.all())
+    tm.assert_raises_regex(TypeError, 'cannot perform any',
+                           lambda: idx.any())
+
+
+def test_boolean_context_compat(idx):
+
+    # boolean context compat
+    def f():
+        if idx:
+            pass
+
```

Comment:
can you change to do this with ``with``

Reply:
I'm not following here.  What is the need to use a context manager here?  None of the functions seems to have a valid __enter__ or __exit__ methods.

Reply:
yep exactly